From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zcbenz <zcbenz@gmail.com>
Date: Thu, 20 Sep 2018 17:50:27 -0700
Subject: disable_detach_webview_frame.patch

Don't detach the frame for webview, we will manage the WebContents
manually.
This is part of the fixes for https://github.com/electron/electron/issues/14211.
We should revisit this bug after upgrading to newer versions of Chrome,
this patch was introduced in Chrome 66.

diff --git a/content/browser/frame_host/render_frame_proxy_host.cc b/content/browser/frame_host/render_frame_proxy_host.cc
index b06a4b9242d30869f9b37fee1dc3ecbe7dcc06d4..b7793aaa094ad342fa15491653baac268a89d524 100644
--- a/content/browser/frame_host/render_frame_proxy_host.cc
+++ b/content/browser/frame_host/render_frame_proxy_host.cc
@@ -286,6 +286,12 @@ void RenderFrameProxyHost::SetDestructionCallback(
 
 void RenderFrameProxyHost::OnDetach() {
   if (frame_tree_node_->render_manager()->ForInnerDelegate()) {
+    // Don't detach the frame for webview, we will manage the WebContents
+    // manually.
+    // We should revisit this bug after upgrading to newer versions of Chrome,
+    // this patch was introduced in Chrome 66.
+    return;
+
     // Only main frame proxy can detach for inner WebContents.
     DCHECK(frame_tree_node_->IsMainFrame());
     frame_tree_node_->render_manager()->RemoveOuterDelegateFrame();
